syntax = "proto3";
service Movie {
  rpc AddMovie(MovieData) returns (MovieData) {}
  rpc DeleteMovie(MovieID) returns (Empty) {}
  rpc UpdateMovieRate(Rate) returns (Empty) {}
  rpc GetMovieByID(MovieID) returns (MovieData) {}
  rpc GetListMovies(Empty) returns (stream MovieData) {}
  rpc GetMovieByDirector(Director) returns (MovieData) {}
  rpc GetMovieByTitle(Title) returns (MovieData) {}
}
message MovieID {
  string id = 1;
}
message Rate {
  float rating = 1;
  string id = 2;
}
message Director {
  string director = 1;
}
message Title {
  string title = 1;
}
message MovieData {
  string title = 1;
  float rating = 2;
  string director = 3;
  string id = 4;
}
message Empty {
}